### ISSUE TITLE
video sprite on ios insecure Operation

### ISSUE THREAD

**Issue by JenGL** [pixijs/pixi.js#4144]
Now that it's possible to play video inline in iOS i'm retrying to use the PIXI video sprite.
But even if with others devices it works perfectly in iOS i encounter this error: 
"SecurityError (DOM Exception 18): The operation is insecure."

here a fiddle: https://jsfiddle.net/220t9e2r/9/

The mp4 file is served with cross origin enabled, but can't made it work on iOS.

Any idea ?

**Comment by englercj** [comment]
Try setting `.crossOrigin` of your video element to `'anonymous'`.

**Comment by JenGL** [comment]
@englercj it is already done. In the html tag it's present the attribute `crossorigin="anonimous"`
In fact this is making the fiddle in android but not on ios

**Comment by ivanpopelyshev** [comment]
Sometimes people write "anonimous". Is that the case for you?

**Comment by englercj** [comment]
Looks like in the jsfiddle it says "anonimous":

```html
<video crossorigin="anonimous" playsinline autoplay controls id="video" style="display: none"></video>
```

Change that to "anonymous":

```html
<video crossorigin="anonymous" playsinline autoplay controls id="video" style="display: none"></video>
```

**Comment by JenGL** [comment]
yes sorry in the fiddle anonymous was miss-written, but even with the correct string the problem is not solved..

https://jsfiddle.net/220t9e2r/18/ in iPhone the html5 video tag is visible, but the video sprite gives the same error: "SecurityError (DOM Exception 18): The operation is insecure."

**Comment by joeredfearn** [comment]
I am actually having the very same issue and I have spelt anonymous correctly.

```javascript
const texture = PIXI.Texture.fromVideo(this.getVideoURL());
texture.baseTexture.on("loaded", (baseTexture) => {
});

const source = texture.baseTexture.source;
source.setAttribute("crossorigin", "anonymous");

this._video = new PIXI.Sprite(texture);
this._video.anchor = new PIXI.Point(0.5, 0.5);
this.addChild(this._video);
```

This code allows the video to stream from a CMS but the same code fails on IOS devices.

```javascript
"SecurityError (DOM Exception 18): The operation is insecure."
```

**Comment by themoonrat** [comment]
I think it might be because you're adding the crossorigin attribute _after_ you've requested to try and load the video?

**Comment by themoonrat** [comment]
Try using Texture.fromVideoUrl() from this version of pixi: https://pixijs.download/video-base-texture-cross-origin/pixi.js

**Comment by joeredfearn** [comment]
thanks for the advice, ill give it a shot and let you know the results!

**Comment by joeredfearn** [comment]
is that just PIXI v4.5.6 or a special fork?

I would like to access all the bin.

**Comment by ivanpopelyshev** [comment]
Its that branch: https://github.com/pixijs/pixi.js/tree/video-base-texture-cross-origin , automatically built, that's how custom versions appear in "pixijs.download". https://github.com/pixijs/pixi.js/compare/video-base-texture-cross-origin

**Comment by joeredfearn** [comment]
ok cool, what is a better way of writing the code above then?

**Comment by themoonrat** [comment]
to be fair, I don't think changes would be needed, as by passing a string to fromVideo, it'll use the fromVideoUrl internally.

```
const texture = PIXI.Texture.fromVideo(this.getVideoURL());
this._video = new PIXI.Sprite(texture);
this._video.anchor.set(0.5);
this.addChild(this._video);
```

should work

**Comment by joeredfearn** [comment]
OK so i have tried that branch and it not fixed the issue.

```javascript
An attempt was made to break through the security policy of the user agent.
```
This is the new error i'm getting. However after doing some digging on the internet I have found out that there is an issue with IOS Safari not supporting CORS properly. This is no longer an issue with IOS 11. I downloaded the latest IOS and the video streams correctly as it should.

This is not really a solution to my problem but its something I guess.
